src/Cli/Commands/ExportOptions.cs

```diff
@@ -1,27 +1,63 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Azure.DataApiBuilder.Core.Generator;
 using CommandLine;
 
 namespace Cli.Commands
 {
-    [Verb("export", isDefault: false, HelpText = "Export the GraphQL schema as a file and save to disk", Hidden = false)]
+    [Verb("export", isDefault: false, HelpText = "Export the GraphQL schema as a file and save it to disk.", Hidden = false)]
     public class ExportOptions : Options
     {
-        public ExportOptions(bool graphql, string outputDirectory, string? config, string? graphqlSchemaFile) : base(config)
+        public ExportOptions(bool graphql, string outputDirectory, string? graphqlSchemaFile,
+                bool? generate, string? samplingMode, int? numberOfRecords, string? partitionKeyPath, int? maxDays, int? groupCount, string? config) : base(config)
         {
             GraphQL = graphql;
             OutputDirectory = outputDirectory;
-            GraphQLSchemaFile = graphqlSchemaFile ?? "schema.graphql";
+            GraphQLSchemaFile = graphqlSchemaFile ?? "schema.gql";
+
+            Generate = generate ?? false;
+            SamplingMode = samplingMode ?? SamplingModes.TopNSampler.ToString();
+            NumberOfRecords = numberOfRecords;
+            PartitionKeyPath = partitionKeyPath;
+            MaxDays = maxDays;
+            GroupCount = groupCount;
         }
 
         [Option("graphql", HelpText = "Export GraphQL schema")]
         public bool GraphQL { get; }
 
-        [Option('o', "output", HelpText = "Directory to save to", Required = true)]
+        [Option('o', "output", HelpText = "Specify the directory where the schema file will be saved. This option is required.", Required = true)]
         public string OutputDirectory { get; }
 
-        [Option('g', "graphql-schema-file", HelpText = "The GraphQL schema file name (default schema.graphql)")]
+        [Option('g', "graphql-schema-file", HelpText = "Specify the filename for the exported GraphQL schema. Default is 'schema.gql'.")]
         public string GraphQLSchemaFile { get; }
+
+        [Option("generate", HelpText = "Generates a schema file from the specified CosmosDB database.")]
+        public bool Generate { get; }
+
+        [Option('m', "sampling-mode", HelpText = "Specifies the sampling mode to use. Available modes include: " +
+                                                 "- TopNSampler: It retrieves a specified number of recent records from a Cosmos DB container, optionally filtering by a maximum number of days." +
+                                                 "- PartitionBasedSampler: It retrieves a specified number of records from a Cosmos DB container by fetching records from each partition based on a specified partition key.The sampling is configurable by the number of records per partition and the time range considered." +
```

Comment:
It retrieves a specified number of records from an Azure Cosmos DB container by fetching records from each partition using a given partition key. The number of records per partition and the time range are configurable.